tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -235,6 +240,25 @@ public:
    /// \return A pointer to the newly created RNTupleProcessor.
    static std::unique_ptr<RNTupleProcessor>
    CreateChain(const std::vector<RNTupleOpenSpec> &ntuples, std::unique_ptr<RNTupleModel> model = nullptr);
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Create a new RNTuple processor for horizontallly concatenated RNTuples.
+   ///
+   /// \param[in] ntuples A list specifying the names and locations of the ntuples to process. The first ntuple in the
+   /// list will be considered the primary ntuple and drives the processor iteration loop. Subsequent ntuples are
+   /// considered auxiliary, whose entries to be read are determined by the primary ntuple (which does not necessarily
+   /// have to be sequential).
+   /// \param[in] joinFields The names of the fields on which to join, in case the specified ntuples are unaligned.
+   /// These join fields have to present in every ntuple specified. If an empty list is provided, it is assumed that
+   /// the specified ntuple are fully aligned, and `RNTupleIndex` will not be used.
+   /// \param[in] models A list of models for the ntuples. This list must either contain a model for each ntuple in
+   /// `ntuples` (following the specification order), or be empty. When the list is empty, the default model (i.e.
+   /// containing all fields) will be used for each ntuple.
+   ///
+   /// \return A pointer to the newly created RNTupleProcessor.
+   static std::unique_ptr<RNTupleProcessor> CreateJoin(const std::vector<RNTupleOpenSpec> &ntuples,
+                                                       const std::vector<std::string> &joinFields,
```

Comment:
I'm wondering if we might want to make the `joinFields` parameter optional, so that by default if events are aligned a user only needs to pass the ntuple spec list. But I understand we might want to be more explicit and make sure users know what they are doing. What is your take on this?

Reply:
I actually changed it to be non-optional after your comment in the first round :-). 

Indeed, the rationale between the non-optional parameter is the fact that this way it has to be explicitly signaled that the RNTuples are aligned. In principle, this signalisation could also be done in another way, e.g. through a dedicated `CreateAlignedJoin` or something. I am not super opnionated on how this is done, but it should be made as difficult as possible to unintentionally wrongly create an aligned join.

Reply:
Fully agree with your statement, and apologies for adding more confusion due to my bad memory :) I believe for this PR it's a sane approach to keep the parameter required!